pkg/csi/service/vanilla/controller.go

```diff
@@ -1917,65 +1886,63 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
 							volumeInfo.VolumeID.Id, vcHost, err)
 					}
 				}
-
-				attributes := make(map[string]string)
-				attributes[common.AttributeDiskType] = common.DiskTypeFileVolume
-
-				resp := &csi.CreateVolumeResponse{
-					Volume: &csi.Volume{
-						VolumeId:      volumeID,
-						CapacityBytes: int64(units.FileSize(volSizeMB * common.MbInBytes)),
-						VolumeContext: attributes,
-					},
-				}
-				return resp, "", nil
+				break
 			}
-			if len(fsEnabledCandidateDatastores) == 0 {
-				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to find  File Service enabled datastores for the vcTopologySegmentsMap: %+v",
-					vcTopologySegmentsMap)
+			// After iterating over all VCs, if volumeID is still empty, we error out.
+			if volumeID == "" {
+				if faultType == "" {
+					faultType = csifault.CSIInternalFault
+				}
+				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to create file volume. Errors encountered: %+v", combinedErrMssgs)
 			}
-		}
-		// When FSS for topology-aware-file-volume is disabled and topologyRequirement is nil, below code is invoked
-		filterSuspendedDatastores := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CnsMgrSuspendCreateVolume)
-		var fsEnabledClusterToDsInfoMap map[string][]*cnsvsphere.DatastoreInfo
-		if multivCenterCSITopologyEnabled {
-			vcHost = c.managers.CnsConfig.Global.VCenterIP
-			fsEnabledClusterToDsInfoMap = c.authMgrs[vcHost].GetFsEnabledClusterToDsMap(ctx)
 		} else {
-			fsEnabledClusterToDsInfoMap = c.authMgr.GetFsEnabledClusterToDsMap(ctx)
-		}
-
-		var filteredDatastores []*cnsvsphere.DatastoreInfo
-		for _, datastores := range fsEnabledClusterToDsInfoMap {
-			filteredDatastores = append(filteredDatastores, datastores...)
-		}
+			// When FSS for topology-aware-file-volume is disabled and topologyRequirement is nil, below code is invoked
```

Comment:
change comment to

When topologyRequirement is nil, below code is invoked

Reply:
Done